### PR TITLE
shop

### PR DESCRIPTION
**Pull request by hsin-yunn**
1. PageCover
2. PageSetting
3. NewsBannerGroup
4. NewsBanner
5. ShopOrder use bonus_points_deduct,disounct_code

### CODE REVIEW

File: src/Controllers/NewsBannerGroupController.php [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+use Wasateam\Laravelapistone\Helpers\ModelHelper;
+
+/**
+ * @group NewsBannerGroup Banner群組
+ *
+ * APIs for news_banner_group
+ *
```
Comment (davidturtle) [inline src/Controllers/NewsBannerGroupController.php:13]: 說明勒

File: src/Resources/NewsBannerGroup.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+   */
+  public function toArray($request)
+  {
+    return [
+      "id"         => $this->id,
+      "created_at" => $this->created_at,
+      "updated_at" => $this->updated_at,
+      "type"       => $this->type,
+      "name"       => $this->name,
+    ];
```
Comment (davidturtle) [inline src/Resources/NewsBannerGroup.php:23]: resource 要分 mode
webapi 盡可能不用 created_at, updatedat

File: src/Resources/PageCover.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  {
+    return [
+      'id'         => $this->id,
+      'name'       => $this->name,
+      'start_date' => $this->start_date,
+      'end_date'   => $this->end_date,
+      'is_active'  => $this->is_active,
+      'link'       => $this->link,
+      'image'      => $this->image,
+    ];
```
Comment (davidturtle) [inline src/Resources/PageCover.php:25]: resource 要分 mode
webapi 盡可能不用 created_at, updatedat, is_active 等

File: src/Resources/PageSetting.php [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+   */
+  public function toArray($request)
+  {
+    return [
+      "id"         => $this->id,
+      "created_at" => $this->created_at,
+      "updated_at" => $this->updated_at,
+      "route"      => $this->route,
+      "name"       => $this->name,
+    ];
```
Comment (davidturtle) [inline src/Resources/PageSetting.php:23]: webapi 盡可能不用 created_at, updatedat, is_active 等

File: src/Resources/ShopProductCoverFrame.php [hunk 5]
```diff
@@ -16,18 +16,25 @@ public function toArray($request)
...
     if (config('stone.mode') == 'cms') {
       $res = [
-        'id'   => $this->id,
-        'name' => $this->name,
-        'url'  => $this->url,
+        'id'         => $this->id,
+        'name'       => $this->name,
+        'url'        => $this->url,
+        'start_date' => $this->start_date,
+        'end_date'   => $this->end_date,
```
Comment (davidturtle) [inline src/Resources/ShopProductCoverFrame.php:23]: cms 盡可能要 created_at, updated_at

File: src/Resources/ShopProductCoverFrame_R1.php [hunk 6]
```diff
@@ -16,9 +16,12 @@ public function toArray($request)
...
       $res = [
-        'id'   => $this->id,
-        'name' => $this->name,
-        'url'  => $this->url,
+        'id'         => $this->id,
+        'name'       => $this->name,
+        'url'        => $this->url,
+        'start_date' => $this->start_date,
+        'end_date'   => $this->end_date,
+        'ordre_type' => $this->order_type,
```
Comment (davidturtle) [inline src/Resources/ShopProductCoverFrame_R1.php:24]: cms 盡可能要 created_at, updated_at

File: src/Controllers/NewsBannerGroupController.php (hunk 1 above)
Comment (hsin-yunn) [reply src/Controllers/NewsBannerGroupController.php:13]: done

File: src/Resources/NewsBannerGroup.php (hunk 2 above)
Comment (hsin-yunn) [reply src/Resources/NewsBannerGroup.php:23]: donw

File: src/Resources/PageCover.php (hunk 3 above)
Comment (hsin-yunn) [reply src/Resources/PageCover.php:25]: done

File: src/Resources/PageSetting.php (hunk 4 above)
Comment (hsin-yunn) [reply src/Resources/PageSetting.php:23]: done

File: src/Resources/ShopProductCoverFrame_R1.php (hunk 6 above)
Comment (hsin-yunn) [reply src/Resources/ShopProductCoverFrame_R1.php:24]: done

File: src/Resources/ShopProductCoverFrame.php (hunk 5 above)
Comment (hsin-yunn) [reply src/Resources/ShopProductCoverFrame.php:23]: done

**Pull request merged**